src/unix.rs

```diff
@@ -28,8 +30,19 @@ pub struct Session {
     tx: RefCell<UnboundedSender<Message>>,
 }
 
+// A type erased handler type to allow sending Easy2 handles across channels.
+// Relies on internals of the `curl` crate.
+// Watch out on auto trait bounds! Currently, `Send` is required due to usage
+// of threads on Windows.
+struct TypeErasedHandler;
+impl Handler for TypeErasedHandler {}
+
 enum Message {
     Execute(Easy, oneshot::Sender<io::Result<(Easy, Option<Error>)>>),
+    Execute2(
+        Easy2<TypeErasedHandler>,
```

Comment:
We need to force a type erase (see the reason below), which relies on (works because) `Easy2`:

1. Boxing the handler (making it immovable)
2. Not touching anything related to the handler in `add2`/`remove2`. In other words, these functions only interact with the raw curl easy pointer.

Reply:
If a bug happens though this will leak information as `Easy2<WrongType>` is dropped, and this is also relying too much I think on the internal memory layout of `Easy2<T>`